### PR TITLE
Fifo depth

### PR DESCRIPTION
**Pull request by GregAC**
This only touched the vendored in IP, the SPI FIFO sizes will be addressed via https://github.com/lowRISC/sonata-system/pull/198.

The I2C is also using a memory back FIFO so I don't think changes there have much impact on FPGA utilization, however I think it's important we have some uniformity in terms of having small FIFOs everywhere.

All builds fine in Vivado, running some tests on the actual board now with the resultant image.

### CODE REVIEW

**Review by marnovandermaas — COMMENTED** [review]
This looks reasonable to me. Not sure what the CI failures are about.

**Comment by marnovandermaas** [comment]
Last force push was a rebase.

**Comment by GregAC** [comment]
> Not sure what the CI failures are about.

Both the UART and I2C drivers only work with the large FIFO depth, fix incoming for this.

**Comment by marnovandermaas** [comment]
Just converting this to draft while the fix for the drivers are in progress.

**Comment by GregAC** [comment]
This will break CI again but wanted to push this update as it contains my fixed I2C test. @alees24 @HU90m PTAL

File: dv/dpi/i2cdpi/i2c_hat_id.cc [hunk 1]
```diff
@@ -11,22 +11,22 @@
...
+  0x72, 0x72, 0x79, 0x20, 0x50, 0x69, 0x53, 0x65,
+  0x6e, 0x73, 0x65, 0x20, 0x48, 0x41, 0x54, 0x3c,  // rry PiSense HAT<
+  0xfc, 0x02, 0x00, 0x01, 0x00, 0x20, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  // ..... ..........
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  // ................
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xed,
+  0x6e, 0x03, 0x00, 0x02, 0x00, 0x0e, 0x00, 0x00,  // ........n.......
+  0x00, 0x72, 0x70, 0x69, 0x2d, 0x73, 0x65, 0x6e,
+  0x73, 0x65, 0x2d, 0x76, 0x32, 0x37, 0x54, 0xff   // .rpi-sense-v27T.
```
Comment (GregAC) [inline dv/dpi/i2cdpi/i2c_hat_id.cc:29]: The previous dump was missing half of the data due to a bug in the I2C driver (which is fixed with https://github.com/lowRISC/cheriot-rtos/pull/8). This new dump is taken from the sense HAT I have and was obtained with the I2C on a raspberry-pi. Some bytes differ in the first half due to the different UUID and CRCs

**Comment by GregAC** [comment]
Driver changes are in: https://github.com/lowRISC/cheriot-rtos/commit/2c4c85b27677897de68ab57c91163c8ca6a77a46

Unfortunately now hitting a failure in the CHERIoT RTOS test suite. Seems the write to stderr hangs which uses UART instance 1. Given stdout on UART 0 works fine and it's identical RTL and drivers this is a little weird. Waiting for a sim run with wave tracing to debug now.

File: dv/dpi/i2cdpi/i2c_hat_id.cc (hunk 1 above)
Comment (marnovandermaas) [reply dv/dpi/i2cdpi/i2c_hat_id.cc:29]: wow, good to have this fixed!

Comment (marnovandermaas) [reply dv/dpi/i2cdpi/i2c_hat_id.cc:29]: The EEPROM test is failing in CI. Is this related to your rework?

**Comment by GregAC** [comment]
Ok Sense HAT on the FPGA CI has a longer length in it's EEPROM header. I think it's one of the older versions that has a full compiled device tree overlay in the eeprom. With the length read from I2C extended (to 1024) this is now passing.

So hopefully this will now pass CI and be good  to go once the SW stuff is reviewed

**Comment by GregAC** [comment]
One final change, the EEPROM test data in the I2C DPI is now what I captured from the FPGA in CI

**Review by HU90m — APPROVED** [review]
Had a quick look through and it looks good! A good quality piece of yak shaving.

**Review by marnovandermaas — APPROVED** [review]
Thanks for getting this sorted. Much more difficult than initially thought. Great to have this in!

**Pull request merged**